ui/components/pagination.js

```diff
@@ -0,0 +1,95 @@
+import { useRouter } from 'next/router'
+import { ChevronLeftIcon, ChevronRightIcon } from '@heroicons/react/solid'
+
+function getPageNums(selected, count, totalPages) {
+  const pageNums = []
+  const beginOffset = Math.max(3, 6 + selected - totalPages)
+  for (
+    let i = Math.max(1, selected - beginOffset);
+    pageNums.length <= count && i <= totalPages;
+    i++
+  ) {
+    pageNums.push(i)
+  }
+
+  return pageNums
+}
+
+function Pages({ path, selected, count, totalPages }) {
+  return getPageNums(selected, count, totalPages).map(page => (
+    <a
+      href={path + '?p=' + page}
+      className={`inline-flex items-center py-2 px-4 text-sm font-medium text-gray-500 hover:text-violet-300 ${
+        selected === page ? 'rounded-md bg-gray-700' : ''
+      }`}
+      key={page}
+    >
+      {page}
+    </a>
+  ))
+}
+
+function LeftArrow({ path, page }) {
+  return (
+    <a
+      href={path + '?p=' + (page > 1 ? page - 1 : 1)}
+      className=' inline-flex items-center pr-1 text-sm font-medium text-gray-500 hover:text-violet-300 '
+    >
+      <ChevronLeftIcon
+        className=' h-5 w-5 text-gray-400 hover:text-violet-300'
+        aria-hidden='true'
+      />
+    </a>
+  )
+}
+
+function RightArrow({ path, page, maxPage }) {
+  return (
+    <a
+      href={path + '?p=' + (page < maxPage ? page + 1 : Math.max(1, maxPage))}
+      className='inline-flex items-center pl-1 text-sm font-medium text-gray-500 hover:text-violet-300 '
+    >
+      <ChevronRightIcon
+        className='ml-3 h-5 w-5 text-gray-400 hover:text-violet-300'
+        aria-hidden='true'
+      />
+    </a>
+  )
+}
+
+export default function Pagination({ curr, totalPages, totalCount }) {
```

Comment:
to check if it is undefined - what you can do is you can set a default value for each props 
so something like this 
`Pagination({ curr = 1, totalPages = 1, totalCount = 0 }) { ... }` 
so then you don't need to do the check on line 65, 66, 67, and 73